data_upload/util/upload_archives.py

```diff
@@ -70,21 +109,38 @@ def upload_file(filename, metadata, nonupload_files, ffpe_samples, level, log):
     return True if 'true' == upload else False
 
 def process_files(config, archive_path, sdrf_metadata, seen_files, nonupload_files, ffpe_samples, level, log):
-    # TODO: set DatafileNameKey and DatafileUploaded here
+    '''
+    process the files in the archive downloaded to the archive_path folder for
+    whether they should be uploaded or not
+    
+    parameters:
+        config: the configuration map
+        archive_path: folder archive was downloaded to and the files extracted to
+        sdrf_metadata: metadata map to update
+        seen_files: files that have been seen in a previously processed archive
+        nonupload_files: list of file extensions of files not to upload
+        ffpe_samples: list of ffpe preserved samples not to upload
+        level: the level of the archive
+        log: logger to log any messages
+    
+    returns:
+        file2metadata: map of filenames to upload with the metadata
+    '''
     files = os.listdir(archive_path)
-    metadatafiles = set(sdrf_metadata.values().keys())
-    archiveonly = files - metadatafiles
-    metaonly = metadatafiles - files
+    metadatafiles = set([curdict.keys()[0] for curdict in sdrf_metadata.values()])

```

Comment:
suggestion: `metadatafiles = set([curdict.keys()[0] for curdict in sdrf_metadata.values() if type(curdict) is dict])`